internal/pkg/cli/project_delete.go

```diff
@@ -0,0 +1,237 @@
+)
+
+type deleteProjOpts struct {
+	*GlobalOpts
+
+	skipConfirmation bool
+	profile          string
+
+	store    projectService
+	deployer deployer
+	ws       workspaceDeleter
+	spinner  progress
+}
+
+type workspaceDeleter interface {
+	Delete() error
+}
+
+func newDeleteProjOpts() (*deleteProjOpts, error) {
+	store, err := store.New()
+	if err != nil {
+		return nil, err
+	}
+
+	ws, err := workspace.New()
+	if err != nil {
+		return nil, err
+	}
+
+	s, err := session.NewProvider().Default()
+	if err != nil {
+		return nil, err
+	}
+	cf := cloudformation.New(s)
+
+	return &deleteProjOpts{
+		GlobalOpts:       NewGlobalOpts(),
+		skipConfirmation: viper.GetBool(yesFlag),
+		profile:          viper.GetString(profileFlag),
+		store:            store,
+		ws:               ws,
+		deployer:         cf,
+		spinner:          termprogress.NewSpinner(),
+	}, nil
+}
+
+func (o *deleteProjOpts) Validate() error {
+	if o.ProjectName() == "" {
+		return errNoProjectInWorkspace
+	}
+
+	return nil
+}
+
+func (o *deleteProjOpts) Ask() error {
+	if o.skipConfirmation {
+		return nil
+	}
+
+	manualConfirm, err := o.prompt.Confirm(
+		fmt.Sprintf(fmtConfirmProjectDeletePrompt, o.ProjectName()),
+		confirmProjectDeleteHelp,
+		prompt.WithTrueDefault())
+
+	if err != nil {
+		return err
+	}
+
+	if !manualConfirm {
+		return errOperationCancelled
+	}
+
+	return nil
+}
+
+func (o *deleteProjOpts) Execute() error {
+	if err := o.deleteApps(); err != nil {
+		return err
+	}
+
+	envs, err := o.store.ListEnvironments(o.ProjectName())
+	if err != nil {
+		return err
+	}
+
+	accounts := newStringSet()
+	regions := newStringSet()
+
+	for _, e := range envs {
+		accounts.put(e.AccountID)
+		regions.put(e.Region)
+	}
+
+	// NOTE: Ordering of project infrastructure stackset and environment deletion is significant.
+	// If the process is cancelled out for any reason before initiating stackset instance deletion
+	// then we're stuck on future rune since we've lost the AccountID and Region information if all the
+	// environments are deleted first.
+	o.spinner.Start("Deleting project resources.")
+	if err := o.deployer.DeleteProject(o.ProjectName(), accounts.items(), regions.items()); err != nil {
+		o.spinner.Stop(log.Serror("Error deleting project resources."))
+
+		return fmt.Errorf("delete project resources: %w", err)
+	}
+	o.spinner.Stop(log.Ssuccess("Deleted project resources."))
+
+	// TODO: move this dependency configuration into newDeleteEnvOpts() function.
+	cfg, err := profile.NewConfig()
+	if err != nil {
+		return err
+	}
+
+	for _, e := range envs {
+		deo := newDeleteEnvOpts()
+		deo.EnvName = e.Name
+		// TODO: enable users to specify a profile per environment deletion?
+		// deo.EnvProfile = o.profile
+		deo.profileConfig = cfg
+		deo.SkipConfirmation = true // always skip sub-confirmations
+		deo.storeClient = o.store
+
+		if err := deo.Ask(); err != nil {
+			return err
+		}
+
+		if err := deo.Validate(); err != nil {
+			return err
+		}
```

Comment:
We should `Validate` before `Ask`, but since we fill the values ourselves I don't think we need to even call `Validate`